Vingette for step by step guide how to perform t-test. (#286)
Olink-Proteomics/OlinkRPackage: OlinkAnalyze/vignettes/study_design_3groups.Rmd

---
title: "A Guide to Comparative Analysis of Muliple Groups with Olink Data"
author: "Olink DS team"
date: "`r Sys.Date()`"
output: 
  html_vignette:
    latex_engine: xelatex
    toc: true
    toc_depth: 2
vignette: >

>   %\VignetteIndexEntry{A Guide to Comparative Analysis of Muliple Groups with Olink Data}

Reviewer: ```suggestion
  %\VignetteIndexEntry{A Guide to Comparative Analysis of Multiple Groups with Olink Data}
```

  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
--- 

```{r setup, include = FALSE}
knitr::opts_chunk$set(
collapse = TRUE,
comment = "#>")
options(tibble.print_min = 4L, tibble.print_max = 4L)
library(dplyr)
library(stringr)
library(ggplot2)
library(OlinkAnalyze)

df <- OlinkAnalyze::npx_data1 %>% 
  filter(!str_detect(SampleID, 'CONTROL'))
df_QC <- OlinkAnalyze::npx_data1 %>% 
  filter(!str_detect(SampleID, 'CONTROL'))

```

# Introduction  

> This vignette is designed to assist users working with 3 or more distinct groups using ANOVA. This guide aims to provide you with a comprehensive overview of the steps involved in analyzing and comparing data from multiple groups in Olink data using the Olink R package.  These groups can encompass a variety of categories, such as disease subtypes, treatment regimens,or genetic variations. The objective of the analysis is to identify for which proteins the groups differ from at least one another. 

Reviewer: ```suggestion
This vignette is designed to assist users working with 3 or more distinct groups using ANOVA. This guide aims to provide you with a comprehensive overview of the steps involved in analyzing and comparing data from multiple groups in Olink data using Olink's R package, Olink Analyze.  These groups can encompass a variety of categories, such as disease subtypes, treatment regimens, or genetic variations. The objective of the analysis is to identify proteins whose expressions differ between at least 2 of the groups. 
```

# Installation 

You can install Olink® Analyze from CRAN which is needed for the following walkthrough.

```{r, eval=FALSE}
install.packages("OlinkAnalyze")
```

After installing the package, the package needs to be loaded. Several packages from the tidyverse suite of packages are also used throughout this vignette, such as dplyr and stringr. Each individual package can be loaded using the _library_ function (`library("package")`) or the entire tidyverse suite of packages can be loaded as shown below.

```{r, eval=FALSE}
library(OlinkAnalyze)
library(tidyverse)
```

# Study overview

This vignette will help you if you have Olink data and are unsure on how to compare your groups. The Olink data file shouldn't be tempered with before loading it into R, as that may cause the data to not be recognized and loaded by the `read_NPX` function. Using the read_NPX function will always give you your data in a long format, where each row represent one sample and one protein. To load in your data in R use the following code:

```{r message=FALSE, eval=FALSE}
data <- read_NPX("~/NPX_file_location.csv")

> # Argument should be the location of your NPX data. Data are usually in parquet, csv, xlsx or zip format depending on the product.

Reviewer: ```suggestion
# Argument should be the location of your NPX data. 
# Data are usually in parquet, csv, xlsx or zip format.
# The data format depends on the product.
```

#the post hoc test should only be performed on proteins which have a significant result in the ANOVA
significant_proteins <- anova_results %>% 
  filter(Threshold == 'Significant') %>% 
  pull(OlinkID)

anova_posthoc_results<-olink_anova_posthoc(df,
                    olinkid_list = significant_proteins, #selecting significant proteins
                    variable = 'Site',
                    effect = 'Site')

```  

The `olink_anova_posthoc` function arrange the results based on adjusted p-value and in this example on the first row the assay CCL18 have a significant difference between Site_B and Site_C of 5.6 NPX. 

```{r message=FALSE, eval=T, echo=FALSE}
anova_posthoc_results %>% head(5)  %>% 
  select(Assay, OlinkID, UniProt, Panel, contrast ,estimate, Adjusted_pval, Threshold) %>% 
  knitr::kable(format = "html",
               booktabs = T,
               linesep = "",
               digits = 4,
               longtable = T,
               row.names = F,
               caption = "**Table 4.** ANOVA post hoc results of the 5 most significant contrasts and proteins. Some coloumns have been left out due to space.",
               label = "manifest")
```  

## Plots

> To illustrate the results from a ANOVA boxplots are a very effective way. These visualizations provide valuable insights into the differences between the groups groups and help in understanding the significance and effect sizes of the observed effects.

Reviewer: ```suggestion
Boxplots are a very effective way to illustrate the results from an ANOVA. These visualizations provide valuable insights into the differences between the groups and help in understanding the significance and effect sizes of the observed effects.
```

> To generate boxplots the `olink_boxplot` function can be used. The thick black horizontal line within each box indicates the median, the colored area - the interquartile range, the ends of the vertical middle line represent the minimum and the maximum values, and the black dots indicate potential assay outliers (beyond the minumum or maximum valuses). Boxplots of the top 6 significant proteins are plotted in **Figure 3**. 

Reviewer: ```suggestion
To generate boxplots, the `olink_boxplot` function can be used. The thick black horizontal line within each box indicates the median, the colored area - the interquartile range, the ends of the vertical middle line represent the minimum and the maximum values, and the black dots indicate potential assay outliers (beyond the minumum or maximum valuses). Boxplots of the top 6 significant proteins are plotted in **Figure 3**. 
```

```{r message=FALSE, eval=T, fig.height = 4, fig.width = 8, fig.align = "center"}
top6Significant<-anova_results %>% 
  filter(Threshold == 'Significant') %>% # only filter out the significant proteins
  head(6) %>%   # just the top 6 for visualization
  pull(OlinkID) # pull out the OlinkID which is needed in olink_boxplot

boxplot_top6<-olink_boxplot(df = df,
              variable = 'Site',
              olinkid_list = top6Significant,
              posthoc_results = anova_posthoc_results) # to illustrate the significant difference between each group the posthoc result can be added as an argument

``` 

```{r fig.cap = fcap3,  echo = T, message=T, eval=T, fig.height = 4, fig.width = 8,fig.align = "center" }
boxplot_top6[[1]]
```

# Saving results

Having the results saved in csv files (or other format) and the plots in pdf (or other format) can making the results easier to work with, especially if the result should be sent to other people without access to R. 

The following code can be used to save the results as a csv file.

```{r message=FALSE, eval=F}

write.table(anova_results,

>             paste0('ANOVAResults.csv'),

Reviewer: do we need the paste0 here?

            sep=';',row.names = F, quote = F)

write.table(anova_posthoc_results,

>             paste0('ANOVA_PostHoc_Results.csv'),

Reviewer: same here. is paste0 needed?
